test/functional/shared/resources/recipe_bicep_test.go

```diff
@@ -254,13 +254,14 @@ func Test_BicepRecipe_ParameterNotDefined(t *testing.T) {
 		Code: "ResourceDeploymentFailure",
 		Details: []step.DeploymentErrorDetail{
 			{
-				Code: recipes.RecipeDeploymentFailed,
-				// NOTE: There is a bug in our error handling for deployements. We return the JSON text of the deployment error inside the message
-				// of our error. This is wrong.
-				//
-				// See: https://github.com/radius-project/radius/issues/6045
-
-				MessageContains: "Deployment template validation failed: 'The template parameters 'a, b' in the parameters file are not valid",
+				Code:            recipes.RecipeDeploymentFailed,
+				MessageContains: "failed to deploy recipe",
```

Comment:
Should we include the complete error message since recipe name and resource type are constants?

Reply:
Added complete error messages